Sources/Build/BuildPlan.swift

```diff
@@ -962,12 +962,14 @@ public final class SwiftTargetBuildDescription {
     }
 
     /// Testing arguments according to the build configuration.
-    private var testingArguments: [String] {
+    private func testingArguments(for target: ResolvedTarget) -> [String] {
+        //return ["-enable-testing"]
         switch buildParameters.configuration {
         case .debug:
             return ["-enable-testing"]
         case .release:
-            if self.buildParameters.enableTestDiscovery {
+            // 👀 not sure
+            if target.type == .test {
```

Comment:
this is a subtle change, but I think *maybe* more correct? in release builds it used to pass `-enable-testing` to *all* targets when enableTestDiscovery was enabled, this changes it to pass it only to test targets. we should consider this same for debug build, but need to confirm correctness.

Reply:
Passing it only for test targets makes sense to me.  Even when supporting testability of executables etc, that wouldn't be finding test cases _declared_ in the executables.  I assume that the previous code was a mistake.

Reply:
I think this change needs to be gated in some way since users might rely on the current behavior for running tests in release mode. For example, in #2918 I discovered that SwiftPM itself relies on the current behavior.

Reply:
@neonichu how do you suggest we gate it? tools-version?

Reply:
Yah, I think tools-version sounds good.

Reply:
In general we do gate things on the tools version, but I though that in the thread (https://forums.swift.org/t/pitch-enable-test-discovery-by-default/36619/23) we said that this one didn't have to be, because it's already gated by the presence or absence of LinuxMain.swift.  That would also prevent people from having to adopt any other manifest changes to use it.

Reply:
@abertelrud yes, I think as a hole the fact that packages with LinuxMain will continue to work as before should be good. that said, we need to decide what to do with this more subtle change, I dont think passing `-enable-testing` always is a good idea

Reply:
Another option could be that explicitly passing `-enable-test-discovery` will retain the behavior of passing `-enable-testing` implicitly, but not passing it explicitly will not. This way we should be able to not break users who rely on the current behavior without opting new users into it.